plugins/user_agent.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package plugins
+
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+const userAgentHeader = "User-Agent"
+
+// CustomUserAgentHandler returns a http request handler that sets a custom user agent to all aws requests
+func CustomUserAgentHandler() request.NamedHandler {
+	return request.NamedHandler{
+		Name: "ECSLocalEndpointsAgentHandler",
+		Fn: func(r *request.Request) {
+			currentAgent := r.HTTPRequest.Header.Get(userAgentHeader)
+			r.HTTPRequest.Header.Set(userAgentHeader,
+				fmt.Sprintf("aws-fluent-bit-plugin (%s) %s", runtime.GOOS, currentAgent))
```

Comment:
Do we want to differentiate the two fluent bit plugins? Looks like right now we will be merging stats from both kinesis and cloudwatch?

Reply:
They make calls to separate APIs- so we can differentiate it based on that. This way we only have to track on user agent.